src/client_side.cc

```diff
@@ -3021,44 +2955,18 @@ ConnStateData::parseTlsHandshake()
         FwdState::Start(clientConnection, sslServerBump->entry, sslServerBump->request.getRaw(), http ? http->al : nullptr);
     } else {
         Must(sslServerBump->act.step1 == Ssl::bumpPeek || sslServerBump->act.step1 == Ssl::bumpStare);
-        startPeekAndSplice();
+        startPeekAndSpliceStep2();
     }
 }
 
-void httpsSslBumpStep2AccessCheckDone(Acl::Answer answer, void *data)
-{
-    ConnStateData *connState = (ConnStateData *) data;
-
-    // if the connection is closed or closing, just return.
-    if (!connState->isOpen())
-        return;
-
-    debugs(33, 5, "Answer: " << answer << " kind:" << answer.kind);
-    assert(connState->serverBump());
-    Ssl::BumpMode bumpAction;
-    if (answer.allowed()) {
-        bumpAction = (Ssl::BumpMode)answer.kind;
-    } else
-        bumpAction = Ssl::bumpSplice;
-
-    connState->serverBump()->act.step2 = bumpAction;
-    connState->sslBumpMode = bumpAction;
-    Http::StreamPointer context = connState->pipeline.front();
-    if (ClientHttpRequest *http = (context ? context->http : nullptr))
-        http->al->ssl.bumpMode = bumpAction;
-
-    if (bumpAction == Ssl::bumpTerminate) {
-        connState->clientConnection->close();
-    } else if (bumpAction != Ssl::bumpSplice) {
-        connState->startPeekAndSplice();
-    } else if (!connState->splice())
-        connState->clientConnection->close();
-}
-
 bool
 ConnStateData::splice()
 {
     // normally we can splice here, because we just got client hello message
+    assert(sslServerBump);
+
+    // Bump procedure finished, reset current bumping step
+    sslServerBump->step = XactionStep::unknown;
```

Comment:
commit 214ffc8a4 adds the XactionStep::tlsBumpDone.
This commitm between others, sets the ConnStateData->sslServerBump.step (which represent the current bumping step) to tlsBumpDone after connection termination because of bumping rules, splicing or after the client-side (and server-side) TLS connection is established.